modules/Silly.py

```diff
@@ -28,8 +28,8 @@ class SillyError(Exception):
         if text.lower().startswith("say "):
             return Response(
                 confidence=4,
-                text=self.utils.modules_dict["Factoids"].dereference(text.partition(" ")[2]) + "!",
-                why="%s told me to say it!" % who,
+                text=self.dereference(text.partition(" ")[2], who) + "!",
```

Comment:
@robertskmiles I know you didn't like the move of `dereference` method from Factoids module to the parent class ... but here is my argument for using statically analyzable code instead of crazy dynamic dispatch in practice - type errors can be found before they happen in production:

![image](https://user-images.githubusercontent.com/1087670/194743099-af8c9aa3-a197-453d-a0a1-87447cbe023a.png)